# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/SharedFontList.h
# Commit: 40fcc0cdc593
# Full Hash: 40fcc0cdc593fa337273631de435a2901ec881f6
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-02-07 04:17:40
# Regressor Bug: 1878286
# File Overlap Count: 2
# Description:
#   Bug 1878286 - Eliminate usage of fontlist::Pointer in IPDL. r=gfx-reviewers,lsalzman
#   
#   No change in behavior.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D200564
# ==============================================================================

diff -r c48ca61bc925 -r 40fcc0cdc593 gfx/thebes/SharedFontList.h
--- a/gfx/thebes/SharedFontList.h	Tue Feb 06 10:15:29 2024 +0000
+++ b/gfx/thebes/SharedFontList.h	Tue Feb 06 10:41:50 2024 +0000
@@ -198,7 +198,8 @@
     return !mDescriptor.IsNull() && mIndex != uint16_t(-1);
   }
 
-  void SetCharacterMap(FontList* aList, gfxCharacterMap* aCharMap);
+  void SetCharacterMap(FontList* aList, gfxCharacterMap* aCharMap,
+                       const Family* aFamily);
 
   String mDescriptor;
   uint16_t mIndex;
@@ -330,6 +331,10 @@
 
   void SetupFamilyCharMap(FontList* aList);
 
+  // Return the index of this family in the font-list's Families() or
+  // AliasFamilies() list, and which of those it belongs to.
+  std::pair<uint32_t, bool> FindIndex(FontList* aList) const;
+
  private:
   // Returns true if there are specifically-sized bitmap faces in the list,
   // so size selection still needs to be done. (Currently only on Linux.)